java/kotlin-extractor/src/main/kotlin/KotlinExtractorExtension.kt

```diff
@@ -201,6 +201,8 @@ class KotlinExtractorGlobalState {
     val syntheticToRealClassMap = HashMap<IrClass, IrClass?>()
     val syntheticToRealFunctionMap = HashMap<IrFunction, IrFunction?>()
     val syntheticToRealFieldMap = HashMap<IrField, IrField?>()
+
+    val deduplicatedWarnings = mutableSetOf<String>()
```

Comment:
Shouldn't these be some variants of concurrent maps? Is there any guarantee that we run single threaded? Could `KotlinExtractorExtension::generate` be invoked by kotlinc from multiple threads?